code/datums/materials/_material.dm

```diff
@@ -0,0 +1,48 @@
+/*! Material datum
+
+Simple datum which is instanced once per type and is used for every object of said material. It has a variety of variables that define behavior. Subtyping from this makes it easier to create your own materials. 
+
+*/
+
+
+/datum/material
+	var/name = "material"
+	var/desc = "its..stuff."
+	//Var that's mostly used by science machines to identify specific materials, should most likely be phased out at some point
+	var/id = "mat"
+	///Base color of the material, is used for greyscale. Item isn't changed in color if this is null.
+	var/color
+	///Base alpha of the material, is used for greyscale icons.
+	var/alpha
+	///Materials "Traits". its a map of key = category | Value = Bool. Used to define what it can be used for.gold
+	var/list/categories = list()
+	///The type of sheet this material creates. This should be replaced as soon as possible by greyscale sheets.
+	var/sheet_type = null
+	///The type of coin this material spawns. This should be replaced as soon as possible by greyscale coins.
+	var/coin_type = null
+	///Icon state override for walls, appeases the WJohn.
+	var/wall_override
+	///Icon state override for floors, appeases the WJohn.
+	var/floor_override
+	///This is a modifier for both force and integrity, and resembles the strength of the material
+	var/strength = 1
+
+///This proc is called when the material is added to an object. mapload exists mostly to prevent the removal of mapped in variables.
+/datum/material/proc/on_applied(atom/source, amount, mapload = FALSE)
+	source.desc += "<br><u>It is made out of [name]</u>."
+	if(!mapload) //Prevent changing mapload icons, to keep colored toolboxes their looks.
```

Comment:
The problem is this makes it not possible to map in properly created things with materials

Reply:
I realized that too, I'm not sure how to work around that to be completely honest so I'll take suggestions.

Reply:
adding a var to atoms like
```
var/material_flags = MATERIAL_NO_COLOR
```
Would be reasonable

Reply:
Aight, I figured atom vars were a big no no so I avoided them as much as possible.

Reply:
You should avoid them but you're genuinely working on a feature for all atoms here.

This var especially is fine since it likely wont be modified ever and so wont take up any extra memory